lib/cellect/server/adapters/panoptes.rb

```diff
@@ -1,76 +1,71 @@
-                           "WHERE \"workflows\".\"id\" IN (#{ names.join(',') })"
-                         end
-            pg.exec(statement).collect do |row|
-              {
-                'id' => row['id'].to_i,
-                'name' => row['id'],
-                'prioritized' => row['prioritized'] == 't',
-                'pairwise' => row['pairwise'] == 't',
-                'grouped' => row['grouped'] == 't'
-              }
-            end
+          PanoptesAssociation::Workflow.select(:id, :prioritized, :pairwise, :grouped)
+            .where(id: names)
+            .map do |row|
+            {
+              'id' => row.id,
+              'name' => "#{row.id}",
+              'prioritized' => row.prioritized,
+              'pairwise' => row.pairwise,
+              'grouped' => row.grouped
+            }
           end
         end
 
         def load_data_for(workflow_name)
-          with_pg do |pg|
-            statement = <<-SQL
-              SELECT sms.id as id, sms.priority as priority, sms.subject_set_id as group_id
-              FROM workflows w
-              JOIN subject_sets ss ON (ss.workflow_id = w.id)
-              JOIN set_member_subjects sms ON (sms.subject_set_id = ss.id)
-              WHERE w.id = #{ workflow_name } AND sms.state = 0
-            SQL
-            pg.exec(statement).collect do |row|
-              {
-                'id' => row['id'].to_i,
-                'priority' => row['priority'].to_f,
-                'group_id' => row['group_id'].to_i
-              }
-            end
+          PanoptesAssociation::SetMemberSubject.joins(:subject_set)
+            .where(subject_sets: { workflow_id: workflow_name },
+                   state: 0)
+            .select(:subject_id, :priority, :subject_set_id)
+            .map do |row|
+            {
+              'id' => row.subject_id,
+              'priority' => row.priority,
+              'group_id' => row.subject_set_id
+            }
           end
         end
 
-        def load_user(workflow_name, id)
-          with_pg do |pg|
-            statement = <<-SQL
-              SELECT set_member_subject_ids FROM user_seen_subjects
-              WHERE user_id = #{ id } AND workflow_id = #{ workflow_name }
-            SQL
-            pg.exec(statement).collect do |row|
-              row['subject_ids'].map(&:to_i)
-            end
+        def load_user(workflow_name, user_id)
+          PanoptesAssociation::UserSeenSubject.where(workflow_id: workflow_name,
+                                user_id: user_id)
+            .select(:subject_ids)
+            .map do |row|
+            row.subject_ids
           end
         end
 
-        def with_pg
-          @pg.with{ |pg| yield pg }
-        end
-
         def connection_options
           {
+            adapter: "postgresql",
             host: ENV.fetch('PG_HOST', '127.0.0.1'),
             port: ENV.fetch('PG_PORT', '5432'),
             dbname: ENV.fetch('PG_DB', 'cellect'),
```

Comment:
Can you add a pool option here that inherits from the global config? https://github.com/zooniverse/cellect_panoptes/blob/9140f6933aac7df17d730ffe2d00f177f57a028a/start#L27